pandas/core/frame.py

```diff
@@ -1978,7 +1978,48 @@ def memory_usage(self, index=True, deep=False):
         return result
 
     def transpose(self, *args, **kwargs):
-        """Transpose index and columns"""
+        """
+        Transpose index and columns.
+
+        Reflect the dataframe over its main diagonal by writing rows as columns
+        and vice-versa.
+
+        Parameters
+        ----------
+        *args
+            These arguments will be passed to
+            :func:`pandas.compat.numpy.function.validate_transpose`.
```

Comment:
@jorisvandenbossche I think standard for this has changed? we don't need to mention the internal functions here

Reply:
Yes, the explanation we are using in other PRs is "Additional keywords have no effect but might be accepted for compatibility with numpy". 

@igorcadelima and you can put both on a single line, like:

```
*args, **kwargs
    Additional keywords have no effect but might be accepted for
    compatibility with numpy
```

Reply:
Thanks for the comments! The reason I didn't just use "Additional keywords have no effect but might be accepted for compatibility with numpy" is because there will be noticeable effects if we pass 'copy=True`.

```
In [1]: import pandas as pd

In [2]: d = {'col1': [1, 2], 'col2': [3, 4]}

In [3]: df = pd.DataFrame(d)

In [4]: df
Out[4]: 
   col1  col2
0     1     3
1     2     4

In [5]: df2 = df.transpose() # or df.T

In [6]: df2
Out[6]: 
      0  1
col1  1  2
col2  3  4

In [7]: df3 = df.transpose(copy=True)

In [8]: df3
Out[8]: 
      0  1
col1  1  2
col2  3  4

In [9]: df['col1'] = 10

In [10]: df
Out[10]: 
   col1  col2
0    10     3
1    10     4

In [11]: df2
Out[11]: 
       0   1
col1  10  10
col2   3   4

In [12]: df3
Out[12]: 
      0  1
col1  1  2
col2  3  4
```

`copy` is the only keyword I have found to have any effect on the transpose operation. What do you think I should do in this case, @jorisvandenbossche @jreback ?

Reply:
But copy is not a numpy keyword? https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.transpose.html 
So I think we should actually just document `copy` as an actual keyword (although it is not in the signature, for technical reasons, and so the validation script will complain about that, but you can ignore that)

Reply:
Not really. What happens is that when `copy=True`, the method [`copy`](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.ndarray.copy.html#numpy.ndarray.copy) is called on the DataFrame's values, as can be confirmed [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L688).

I just incorporated your comments.